server/src/main/java/org/opensearch/cluster/metadata/IndexMetadata.java

```diff
@@ -243,6 +243,21 @@ static Setting<Integer> buildNumberOfShardsSetting() {
         Property.IndexScope
     );
 
+    /**
+     * Setting to control the number of search only replicas for an index.
+     * A search only replica has the following properties:
+     * 1. Is not primary eligible
+     * 2.
+     */
+    public static final String SETTING_NUMBER_OF_SEARCH_REPLICAS = "index.number_of_search_only_shards";
```

Comment:
Should probably be something like "number of search replicas". We refer to replicas as "replicas", not "replica shards", so we should probably be consistent here.

Reply:
updated to "number_of_search_only_replicas"